File: reorder_leadership.js
const fs = require('fs');
const path = require('path');

// Read the file with UTF-16LE encoding
const filePath = path.join(__dirname, 'temp_past_team.html');

console.log('Reading file...');
const content = fs.readFileSync(filePath, 'utf16le');

console.log(`File size: ${content.length} characters`);

// Search for Dominic
const searchTerm = 'dominic';
const lowerContent = content.toLowerCase();
const idx = lowerContent.indexOf(searchTerm);

if (idx !== -1) {
    console.log(`\nFound '${searchTerm}' at position ${idx}`);
    console.log('\nContext around the match:');
    console.log('='.repeat(80));
    const start = Math.max(0, idx - 500);
    const end = Math.min(content.length, idx + 500);
    console.log(content.substring(start, end));
    console.log('='.repeat(80));
    
    // Try to find the structure pattern
    // Look for card/div patterns that might contain team members
    const contextStart = Math.max(0, idx - 2000);
    const contextEnd = Math.min(content.length, idx + 2000);
    const largeContext = content.substring(contextStart, contextEnd);
    
    // Look for div or section tags that might define a team member card
    const cardPatternMatch = largeContext.match(/<div[^>]*class="[^"]*(?:card|member|team|person)[^"]*"[^>]*>[\s\S]*?dominic[\s\S]*?<\/div>/i);
    
    if (cardPatternMatch) {
        console.log('\n\nPossible card structure found:');
        console.log('='.repeat(80));
        console.log(cardPatternMatch[0].substring(0, 1000));
        console.log('='.repeat(80));
    }
} else {
    console.log(`\n'${searchTerm}' not found in the file`);
}

// Also search for common section markers
console.log('\nSearching for leadership section markers...');
const markers = ['leadership', 'team', 'president', 'executive', 'our team', 'our leadership'];

markers.forEach(marker => {
    const markerIdx = lowerContent.indexOf(marker.toLowerCase());
    if (markerIdx !== -1) {
        console.log(`\nFound '${marker}' at position ${markerIdx}`);
    }
});

File: reorder_leadership.py
#!/usr/bin/env python3
"""
Script to reorder the leadership section in temp_past_team.html
Moves Ezeani Dominic Tochukwu to the first position.
"""

import re

# Read the file with UTF-16LE encoding
file_path = r'c:\Users\pharm victor\Desktop\Coding\ispor_hoer\temp_past_team.html'

with open(file_path, 'r', encoding='utf-16-le') as f:
    content = f.read()

# Find the leadership section
# Look for patterns that might contain team member cards
print("Searching for 'dominic' (case-insensitive)...")
idx = content.lower().find('dominic')
if idx != -1:
    print(f"Found 'dominic' at position {idx}")
    print("\nContext around the match:")
    print("=" * 80)
    print(content[max(0, idx-500):idx+500])
    print("=" * 80)
else:
    print("'dominic' not found in the file")
    
# Also search for common leadership section markers
print("\nSearching for leadership section markers...")
for marker in ['leadership', 'team', 'president', 'executive']:
    idx = content.lower().find(marker)
    if idx != -1:
        print(f"\nFound '{marker}' at position {idx}")
